### DISCUSSION TITLE
UE5 compatiblity 5.0-5.3 - introduce namespace?

### DISCUSSION THREAD

**Discussion by tsky1971** [JSBSim-Team/jsbsim#1053] [General]
Hi,
I don't see full compatibility from UE 5.0 to 5.3. Each version seems to change many details. Problems only arise during deployment. Things work in the editor. But everyone actually wants to create a deployment.

5.1 Forces categorization of BP declarations, the IWYU concept becomes strict.

5.2 seems to have at least one naming conflict issue:
`C:\Program Files\Epic Games\UE_5.2\Engine\Plugins\Experimental\UserToolBoxBasicCommand\Source\UserToolBoxBasicCommand\Public\EngineCommand.h(12): Error: Class 'UEngineCommand' shares engine name 'EngineCommand' with struct 'FEngineCommand' in D:\UE_Github\tsky1971\jsbsim\UE\5.2\JSBSimFlightDynamicsModel\HostProject\Plugins\JSBSimFlightDynamicsModel\Source\JSBSimFlightDynamicsModel\Public\FDMTypes.h(140)`

With 5.3 I'm currently having problems getting a JoystickPlugin compatible for deployment. So I expect one or two problems here too.

I suggest introducing a namespace for the system. This would also avoid many problems during deployment by using similar naming in other installed plugins.

Maybe I missed something or have a configuration with a local problem?

T.

**Comment by gallonmate** [comment]
The last I tested the default project as a packaged game was with UE5.2.0, using the launcher engine version and the package built as "development build". Maybe other build types or engine source build shows other errors. Fixes are welcome.

I'm not familiar with namesapces. Would your suggestion be able to meet the UE coding standards for namespace and solve the naming conflicts you encountered? https://docs.unrealengine.com/5.3/en-US/epic-cplusplus-coding-standard-for-unreal-engine/#namespaces

**Comment by tsky1971** [comment]
I read a little and seems with the ustruct etc. it is not possible. So i think good old prefixing it the best possible method.  

![image](https://github.com/JSBSim-Team/jsbsim/assets/7058122/dbdf7914-93cb-49ec-99b8-6cd0f0b18c8c)

I try to prefix all in FDMTypes.h with JSBSim as in the MovementComp, and this works. It´s a litte bit ugly ;-)

T.

**Reply by tsky1971** [reply]
Hm, side effect: I have to recreate the accessor in the BP. Compiling means not working...
